test/e2e/cert_rotation.go

```diff
@@ -72,40 +57,33 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
 	}
 	for _, secretName := range secretList {
 		err := clientSet.CoreV1().Secrets(ironicNamespace).Delete(ctx, secretName, metav1.DeleteOptions{})
-		Expect(err).ToNot(HaveOccurred(), "Cannot detele this secret: %s", secretName)
+		Expect(err).ToNot(HaveOccurred(), "cannot delete this secret: %s", secretName)
 	}
 
-	By("Wait for containers in the ironic pod to be restarted")
+	By("Wait until secrets are recreated")
 	Eventually(func() error {
-		ironicPod, err := getPodFromDeployment(ctx, clientSet, ironicDeployment, ironicNamespace)
-		if err != nil {
-			return err
-		}
-		// check for container in containerNumRestart list
-		for container := range containerNumRestart {
-			notFound := true
-			for _, ironicContainer := range ironicPod.Status.ContainerStatuses {
-				if ironicContainer.Name == container {
-					notFound = false
-					break
-				}
-			}
-			if notFound {
-				return fmt.Errorf("%s container does not exist in Ironic pod", container)
-			}
+		ironicSecretCreated := isSecretRecreated(ctx, clientSet, ironicNamespace, secretList[0])
+		mariaDBSecretCreated := true
+		if mariadbEnabled {
+			mariaDBSecretCreated = isSecretRecreated(ctx, clientSet, ironicNamespace, secretList[1])
 		}
-		if ironicPod.Status.Phase == corev1.PodRunning {
-			for _, container := range ironicPod.Status.ContainerStatuses {
-				if oldNumRestart, exist := containerNumRestart[container.Name]; exist {
-					if !(oldNumRestart < container.RestartCount) {
-						return fmt.Errorf("%s is not restarted", container.Name)
-					}
-				}
-			}
+		if ironicSecretCreated && mariaDBSecretCreated {
 			return nil
 		}
-		return errors.New("ironic pod is not in running state")
+		return errors.New("secret is being recreated")
 	}, input.E2EConfig.GetIntervals(input.SpecName, "wait-pod-restart")...).Should(BeNil())
+
+	// TODO(Sunnatillo): Further extend the test with querying ironic endpoint with new certificates
+	By("Check if all containers are running in ironic pod")
+	Consistently(func() error {
+		ironicPod, err := getPodFromDeployment(ctx, clientSet, ironicDeployment, ironicNamespace)
+		Expect(err).ToNot(HaveOccurred(), "cannot get ironic from Deployment")
+
+		if areAllContainersRunning(ironicPod) {
```

Comment:
I think we can check directly if the deployment is running ?

Reply:
Deployment can show running even though not all containers are running in a pod. I think we should check all the containers.